### PR TITLE
Add automated scripts for installfest and update INSTALLFEST.md

### PR DESCRIPTION
**Pull request by scottyscripts**
I automated everywhere that I could and updated the INSTALLFEST.md to reflect changes.

I removed the sections from INSTALLFEST.md for installing rbenv, Ruby, and postgres, but still included the scripts in config/

Fixed Atom autocomplete issue we discussed, will close issue after these changes are merged

### CODE REVIEW

File: config/atom.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Comment (gaand) [inline config/atom.sh:1]: Is there an issue about using cask to install Atom?  Would that make it possible to have a more comprehensive script?  Also, where's a check to see if the commands are available?

File: config/bash.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  if grep -Fxq "test -f ~/.bashrc && source ~/.bashrc" ~/.bash_profile
+  then
+    echo "Looks good!"
+  else
+    echo "Oh No! It looks like there was an issue. Ask for assistance before continuing.
+    Press [ENTER] to continue."
+    read
+  fi
+
+  echo '/\/usr\/local\/bin/\nd\nwq' | sudo ed /etc/paths
```
Comment (gaand) [inline config/bash.sh]: There's definitely an issue saying not to do this anymore.

File: config/git.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+if which git; then
```
Comment (gaand) [inline config/git.sh]: This test is insufficient. Git is installed on macOS by default.

File: config/npm.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+npm install --global npm
+npm install --global jshint jsonlint grunt-cli remark-lint jscs bower phantomjs-prebuilt
```
Comment (gaand) [inline config/npm.sh]: This is not the correct set of global packages.

File: config/nvm.sh [hunk 5]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+if uname -s != 'Darwin'; then
+  export NVM_DIR="/home/$(whoami)/.nvm"
```
Comment (gaand) [inline config/nvm.sh]: I believe this is unnecessary if nvm is installed and in the path.

File: config/nvm.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+else
+  export NVM_DIR=~/.nvm
```
Comment (gaand) [inline config/nvm.sh]: As the above

File: config/postgres.sh [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+if uname -s != 'Darwin'; then
+  sudo apt-get install postgresql libpq-dev
+  sudo -u postgres createuser `whoami` -s
+
+  sudo -u postgres createdb `whoami`
+else
+  brew install postgres
+  mkdir -p ~/Library/LaunchAgents
```
Comment (gaand) [inline config/postgres.sh]: Is this actually necessary?

File: config/rbenv.sh [hunk 8]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+if which rvm; then
```
Comment (gaand) [inline config/rbenv.sh:2]: Destructive actions should warn the user

File: config/xcode.sh [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+if which xcode-select; then
```
Comment (gaand) [inline config/xcode.sh]: I'm not sure why this would work.

**Review by gaand — CHANGES_REQUESTED** [review]
Why are there separate git scripts?
Are we using linux brew for linux?

Please have a look and let me know if you have questions.

**Comment by scottyscripts** [comment]
I consolidated git_config.sh with git.sh.

After researching the export NVM_DIR, I continually found issues with nvm command not working unless you do the export

removing rvm now prompts the user for a (y/n)

There are many open issues. I commented and addressed the ones I saw as imperative for Friday, but will be clearing the rest.

**Comment by scottyscripts** [comment]
Tested all scripts, fixing any errors encountered. Consolidated two of the git scripts. Looked more into installing atom with cask, and I'm not sure that it would speed up the process. Also edited the INSTALLFEST.md doc to reflect changes.

File: INSTALLFEST.md [hunk 10]
```diff
@@ -266,411 +164,80 @@ brew install nvm
...
+***Note: From this point forward you will open Atom from the command line***
 
-### Install [`hub`](https://github.com/github/hub)
+### OSX ONLY
 
-> hub is a command line tool that wraps git in order to extend it with extra
-> features and commands that make working with GitHub easier. -- [`hub`
-> README](https://github.com/github/hub).
+1.  Download Atom from [Atom.io](https://atom.io/).
+1.  Move Atom from downloads to the Application directory.
```
Comment (jrhorn424) [inline INSTALLFEST.md]: This step is very important. Without it, the command line tool installation will claim to work when it does not actually work. This is due to sandboxing on macOS.

File: config/atom.sh (hunk 1 above)
Comment (jrhorn424) [reply config/atom.sh:1]: > Also, where's a check to see if the commands are available?

I agree, this should be added before merge IMO.

File: config/nvm.sh [hunk 11]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+if [[ $(uname -s) != 'Darwin' ]]; then
+  export NVM_DIR="/home/$(whoami)/.nvm"
+  [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+else
+  export NVM_DIR=~/.nvm
+  source $(brew --prefix nvm)/nvm.sh
+fi
+
+nvm install --lts=boron || nvm install 6.9.4
```
Comment (jrhorn424) [inline config/nvm.sh:10]: Boron is now at 6.10.2. I'm not sure what we want to do about that @gaand.

File: config/rbenv.sh [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+else
+  brew install ruby-build
+fi
+
+rbenv --version
+read -p "# expecting message 'rbenv 1.1.0'
+# if you do not get the message 'rbenv 1.1.0' refer to an instructor \n
+Press [ENTER] to continue"
+
+echo bundler > ~/.rbenv/default-gems
```
Comment (gaand) [inline config/rbenv.sh]: Destructive.

File: config/rbenv.sh [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+rbenv rehash
+
+if [[ $(uname -s) != 'Darwin' ]]; then
+  git clone https://github.com/sstephenson/ruby-build.git ~/.rbenv/plugins/ruby-build
+else
+  brew install ruby-build
+fi
+
+rbenv --version
+read -p "# expecting message 'rbenv 1.1.0'
```
Comment (gaand) [inline config/rbenv.sh]: Check this directly, don't prompt

**Review by gaand — CHANGES_REQUESTED** [review]
Linux brew is important.

File: config/atom.sh (hunk 1 above)
Comment (scottyscripts) [reply config/atom.sh:1]: Cask will only work for macOSX. Tested process in Ubuntu machine

File: config/rbenv.sh (hunk 8 above)
Comment (scottyscripts) [reply config/rbenv.sh:2]: Now prompt user for (y/n). Only remove if yes

File: INSTALLFEST.md (hunk 10 above)
Comment (scottyscripts) [reply INSTALLFEST.md]: I moved these steps and the atom part to end of installfest. cask will work to install on mac, but not linux

File: config/rbenv.sh (hunk 13 above)
Comment (scottyscripts) [reply config/rbenv.sh]: This is now automated

**Comment by scottyscripts** [comment]
Looked into cask for installing atom, but it only works on mac. I updated INSTALLFEST.md to tell them to run `brew cask install atom` for mac (kept old command for linux). I will test scripts again tonight on linux and mac just to make sure we're all good for tomorrow. 

Let me know if anything is unclear.

**Pull request merged**